tests/wake-format/basic/basic.wake

```diff
@@ -599,3 +599,15 @@ export tuple a +++ b =
 
 export tuple (z: Banger) =
   export X: Integer
+
+export data List a =
+    # The empty list. Nil represents a list with no elements.
+    Nil
+    # The prepend constructor. This is often called "cons".
+    #
+    # Examples:
+    # ```
+    #   10, list       # Add a number to `list`
+    #   1, 2, 3, Nil   # Create a list of 3 elements
+    # ```
+    (head: a), (tail: List a) # comment
```

Comment:
We don't have any test for a data constructor with a trailing comment but *not* a leading one, but I don't expect that to affect anything.

Reply:
added more cases!